lib/openstack_query/api/query_api.py

```diff
@@ -199,4 +206,79 @@ def to_html(
         :param groups: a list group to limit output by
         :param kwargs: kwargs to pass to generate table
         """
-        return self.output.to_html(self._query_results, title, groups, **kwargs)
+        _, selected_results = self.executer.parse_results()
+        return self.output.to_html(selected_results, title, groups, **kwargs)
+
+    def then(self, query_type: Union[str, QueryTypes], forward_outputs: bool = True):
+        """
+        Public method to chain current query into another query of a different type
+        and return the new query so that it will work only on the results of the original query.
+        NOTE - query must be run first for this to work
+        NOTE - a shared common property must exist between this query and the new query
+            - i.e. both ServerQuery and UserQuery share the 'USER_ID' property so chaining is possible
+                - see Mappings for more chaining options
+
+        :param query_type: an enum representing the new query to chain into
+        :param forward_outputs: (default=True) boolean that if true - will forward outputs from t
+        his query (and previous chained queries) onto new query.
+            NOTE: You will NOT be able to group/sort by these properties in the new query
+        """
+        if isinstance(query_type, str):
+            query_type = QueryTypes.from_string(query_type)
+
+        link_props = self.chainer.get_link_props(query_type)
+        if not link_props:
+            raise QueryChainingError(
+                f"Query Chaining Error: Could not find a way to chain current query into {query_type}"
+            )
+
+        curr_query_results = self.group_by(link_props[0]).to_list()
+        if not curr_query_results:
+            raise QueryChainingError(
+                "Query Chaining Error: No values found after running query set in {curr_query_obj_cls} aborting. "
+                "Have you run the query first?"
+            )
```

Comment:
If we're going to ask the user to run the query, why not just emit a warning and run it on their behalf?

Reply:
which cloud account do we use? The user has to provide that info somehow - right now we only accept that at `run()`

Reply:
underlying question here is:

1. Do we ask for the `cloud_account` upon creating the query object 
     - so the query is set only to work with those credentials (making it a little inflexible)
     
2. Or do we add `cloud_account` param to all possible methods like; 
    - `to_list`, `to_html`, `to_string`, `then` and possibly more as they get created

3. Or do we just force users to run `run()` first

something to think about

Reply:
Ah I'd forgotten about the cloud account at the run portion, let's leave it as a limitation then :+1: